main.js

```diff
@@ -0,0 +1,69 @@
+// Étape 1
+const pokemonData = require("./pokedex.json");
+
+function makeObjects(data) {
+  let list = [];
+  for (let i = 0; i < data.pokemon.length; i++) {
+    list.push(data.pokemon[i]);
+  }
+  return list;
+}
+
+// Étape 2
+function countPokemon(data) {
+  console.log(data.length);
+}
+
+function findHeavyPokemon(data) {
+  let heavyPokemon = [];
+  for (let i = 0; i < data.length; i++) {
+    let weight = data[i].weight;
+    let formattedWeight = parseFloat(weight.split(" ")[0]);
+    if (formattedWeight > 10) {
+      heavyPokemon.push(data[i].name);
+    }
+  }
+  console.log(heavyPokemon);
+}
+
+function orderByWeight(data) {
```

Comment:
Il y a pas mal d'idée dans cette fonction et ton implémentation corresponds [au tri par insertion](https://fr.wikipedia.org/wiki/Tri_par_insertion). Un axe d'optimisation possible serait d'utiliser `while` plutôt que `for`. En effet, dans ta deuxième boucle, il s'agit, non pas d'itérer sur **tout** les élèments de ton tableau, mais de trouver le pokémon le plus lourd. Cela peut être un travail de factorisation intéressant. Mais sinon c'est super :+1: 

Si tu arrives